lib/features/composer/domain/usecases/upload_attachment_interactor.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/composer/domain/repository/composer_repository.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/upload_attachment_state.dart';
+import 'package:tmail_ui_user/features/login/domain/repository/credential_repository.dart';
+
+class UploadAttachmentInteractor {
+  final ComposerRepository _composerRepository;
+  final CredentialRepository _credentialRepository;
+
+  UploadAttachmentInteractor(this._composerRepository, this._credentialRepository);
+
+  Stream<Either<Failure, Success>> execute(FileInfo fileInfo, AccountId accountId, Uri uploadUrl) async* {
+    try {
+      final result = await Future.wait(
+          [_credentialRepository.getUserName(), _credentialRepository.getPassword()],
```

Comment:
IMO, no need this. This process execute directly in the foreground (where the dio is set up completely) so I think `AuthorizationInterceptor did his work.